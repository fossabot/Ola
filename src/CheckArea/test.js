import React from 'react'
import { CheckArea } from './'
import renderer from 'react-test-renderer'

const options = [
  {
    label: 'Option 1',
    value: 1,
  },
  {
    label: 'Option 2',
    value: 2,
  },
  {
    label: <strong>Option 3</strong>,
    value: 3,
  }
]

it('Default checkarea', () => {
  const tree = renderer
    .create(<CheckArea name="test" options={options}></CheckArea>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Checkarea type checkbox', () => {
  const tree = renderer
    .create(<CheckArea name="test" options={options} type="checkbox"></CheckArea>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Checkarea type radio with value="2"', () => {
  const tree = renderer
    .create(<CheckArea name="test" options={options} type="checkbox" value="2"></CheckArea>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Checkarea type radio with value=2', () => {
  const tree = renderer
    .create(<CheckArea name="test" options={options} type="checkbox" value={2}></CheckArea>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
