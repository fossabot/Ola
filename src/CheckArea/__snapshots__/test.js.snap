// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkarea type checkbox 1`] = `
<div
  className="ola_checkArea is-row"
  onChange={[Function]}
  role="radiogroup"
>
  <label
    className="ola_check"
  >
    <input
      checked={null}
      className="ola_check-input"
      name="test"
      type="checkbox"
      value={1}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          Option 1
        </div>
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={null}
      className="ola_check-input"
      name="test"
      type="checkbox"
      value={2}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          Option 2
        </div>
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={null}
      className="ola_check-input"
      name="test"
      type="checkbox"
      value={3}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          <strong>
            Option 3
          </strong>
        </div>
      </div>
    </div>
  </label>
</div>
`;

exports[`Checkarea type radio with value="2" 1`] = `
<div
  className="ola_checkArea is-row"
  onChange={[Function]}
  role="radiogroup"
>
  <label
    className="ola_check"
  >
    <input
      checked={false}
      className="ola_check-input"
      name="test"
      type="checkbox"
      value={1}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          Option 1
        </div>
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={true}
      className="ola_check-input"
      name="test"
      type="checkbox"
      value={2}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          Option 2
        </div>
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={false}
      className="ola_check-input"
      name="test"
      type="checkbox"
      value={3}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          <strong>
            Option 3
          </strong>
        </div>
      </div>
    </div>
  </label>
</div>
`;

exports[`Checkarea type radio with value=2 1`] = `
<div
  className="ola_checkArea is-row"
  onChange={[Function]}
  role="radiogroup"
>
  <label
    className="ola_check"
  >
    <input
      checked={false}
      className="ola_check-input"
      name="test"
      type="checkbox"
      value={1}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          Option 1
        </div>
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={true}
      className="ola_check-input"
      name="test"
      type="checkbox"
      value={2}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          Option 2
        </div>
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={false}
      className="ola_check-input"
      name="test"
      type="checkbox"
      value={3}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          <strong>
            Option 3
          </strong>
        </div>
      </div>
    </div>
  </label>
</div>
`;

exports[`Default checkarea 1`] = `
<div
  className="ola_checkArea is-row"
  onChange={[Function]}
  role="radiogroup"
>
  <label
    className="ola_check"
  >
    <input
      checked={null}
      className="ola_check-input"
      name="test"
      type="radio"
      value={1}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          Option 1
        </div>
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={null}
      className="ola_check-input"
      name="test"
      type="radio"
      value={2}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          Option 2
        </div>
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={null}
      className="ola_check-input"
      name="test"
      type="radio"
      value={3}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        <div
          className="ola_checkArea-content"
        >
          <strong>
            Option 3
          </strong>
        </div>
      </div>
    </div>
  </label>
</div>
`;
